src/erlcloud_aws.erl

```diff
@@ -165,12 +165,13 @@ aws_request4_no_update(Method, Protocol, Host, Port, Path, Params, Service,
                        Headers, #aws_config{} = Config) ->
     Query = erlcloud_http:make_query_string(Params),
     Region = aws_region_from_host(Host),
+    Uri = erlcloud_http:url_encode_loose(Path),
```

Comment:
one more question - which service api caused issues for you? 
in S3 the request is formed/signed other way and it works there.